commons-numbers-fraction/src/main/java/org/apache/commons/numbers/fraction/Fraction.java

```diff
@@ -775,19 +778,45 @@ public Fraction divide(Fraction value) {
      */
     @Override
     public Fraction pow(final int exponent) {
+        if (exponent == 1) {
+            return this;
+        }
         if (exponent == 0) {
             return ONE;
         }
-        if (isZero()) {
-            return ZERO;
+        if (exponent == -1) {
+            return this.reciprocal();
         }
-
-        if (exponent < 0) {
-            return new Fraction(ArithmeticUtils.pow(denominator, -exponent),
-                                ArithmeticUtils.pow(numerator,   -exponent));
+        if (this.denominator == 0) {
+            if (exponent > 0) {
+                throw new ArithmeticException(STRING_THE_DENOMINATOR_MUST_NOT_BE_ZERO);
+            } else {
+                return ZERO;
+            }
+        }
+        if (this.numerator == 0) {
+            if (exponent < 0) {
+                throw new ArithmeticException(STRING_THE_DENOMINATOR_MUST_NOT_BE_ZERO);
+            } else {
+                return ZERO;
+            }
         }
-        return new Fraction(ArithmeticUtils.pow(numerator,   exponent),
-                            ArithmeticUtils.pow(denominator, exponent));
+        if (exponent > 0) {
+            return new Fraction(
+                    ArithmeticUtils.pow(this.numerator, exponent),
+                    ArithmeticUtils.pow(this.denominator, exponent)
+            );
+        }
+        if (exponent == Integer.MIN_VALUE) {
+            return new Fraction(
+                    ArithmeticUtils.pow(this.denominator, Integer.MAX_VALUE) * this.denominator,
```

Comment:
A raw multiplication here is a bad idea. You should at least use Math.multipyExact. For this extreme edge case I think some variant of the following simpler logic is clearer:
```java
if (exponent == Integer.MIN_VALUE) {
      // MIN_VALUE can't be negated
      final int temp = exponent / 2;
      return new Fraction(ArithmeticUtils.pow(denominator, -temp),
                          ArithmeticUtils.pow(numerator,   -temp)).pow(2);
}
```

Reply:
@aherbert 
No need to use ArithmeticUtils.pow here.
Because:
If this.denominator be 0,1,or -1, then it will never overflow.
Other wise, any other int do pow with Integer.MAX_VALUE will overflow, it will overflow before we do the multiply.
So here is a tricky but correct implement.